src/commands/check/index.ts

```diff
@@ -17,57 +26,117 @@ interface Ctx {
   static description = "Check installed package versions and compatibility";
 
+  static flags = {
+    print: Flags.string({
+      char: "p",
+      description: "File to write output to",
+    }),
+  };
+
   public async run(): Promise<void> {
+    const { flags } = await this.parse(Check);
+    const swankyNodeVersion = this.swankyConfig.node.version;
+    const isSwankyNodeInstalled = !!swankyNodeVersion;
+    const anyContracts = Object.keys(this.swankyConfig?.contracts).length > 0;
     const tasks = new Listr<Ctx>([
+      {
+        title: "Check OS",
+        task: async (ctx, task) => {
+          ctx.os.platform = process.platform;
+          ctx.os.architecture = process.arch;
+          const supportedPlatforms = ["darwin", "linux"];
+          const supportedArch = ["arm64", "x64"];
+
+          if (!supportedPlatforms.includes(ctx.os.platform)) {
+            throw new Error(`Platform ${ctx.os.platform} is not supported`);
+          }
+          if (!supportedArch.includes(ctx.os.architecture)) {
+            throw new Error(`Architecture ${ctx.os.architecture} is not supported`);
+          }
+
+          task.title = `Check OS: '${ctx.os.platform}-${ctx.os.architecture}'`;
+        },
+        exitOnError: false,
+      },
       {
         title: "Check Rust",
-        task: async (ctx) => {
-          ctx.versions.tools.rust = await commandStdoutOrNull("rustc --version");
+        task: async (ctx, task) => {
+          ctx.versions.tools.rust = (await commandStdoutOrNull("rustc --version"))?.match(/rustc (.*) \((.*)/)?.[1];
+          if (!ctx.versions.tools.rust) {
+            throw new Error("Rust is not installed");
+          }
+          task.title = `Check Rust: ${ctx.versions.tools.rust}`;
         },
+        exitOnError: false,
       },
       {
         title: "Check cargo",
-        task: async (ctx) => {
-          ctx.versions.tools.cargo = await commandStdoutOrNull("cargo -V");
+        task: async (ctx, task) => {
+          ctx.versions.tools.cargo = (await commandStdoutOrNull("cargo -V"))?.match(/cargo (.*) \((.*)/)?.[1];
+          if (!ctx.versions.tools.cargo) {
+            throw new Error("Cargo is not installed");
+          }
+          task.title = `Check cargo: ${ctx.versions.tools.cargo}`;
         },
+        exitOnError: false,
       },
       {
         title: "Check cargo nightly",
-        task: async (ctx) => {
-          ctx.versions.tools.cargoNightly = await commandStdoutOrNull("cargo +nightly -V");
+        task: async (ctx, task) => {
+          ctx.versions.tools.cargoNightly = (await commandStdoutOrNull("cargo +nightly -V"))?.match(/cargo (.*)-nightly \((.*)/)?.[1];
+          if (!ctx.versions.tools.cargoNightly) {
+            throw new Error("Cargo nightly is not installed");
+          }
+          task.title = `Check cargo nightly: ${ctx.versions.tools.cargoNightly}`;
         },
+        exitOnError: false,
       },
       {
         title: "Check cargo dylint",
```

Comment:
where `dylint` is actually used when working with swanky?

Reply:
Nowhere directly, `dylint` has been in place for some time for smart contract projects using swanky. I believe it aligns with the recommended best practices for installation as outlined in the [cargo-contract README](https://github.com/paritytech/cargo-contract/blob/744babfcd61fd74d2e02d20b25b9823ecf18308a/README.md?plain=1#L45).